src/main/java/org/apache/maven/plugins/gpg/GpgVersion.java

```diff
@@ -32,54 +32,46 @@
  */
 public class GpgVersion implements Comparable<GpgVersion>
 {
-    private final String rawVersion;
 
-    private GpgVersion( String rawVersion )
-    {
-        this.rawVersion = rawVersion;
-    }
+    private static final Pattern VERSION_PATTERN = Pattern.compile( "(\\d+\\.)+(\\d+)" );
```

Comment:
@Syquel 
@elharo 
How many times these patterns exist in the project?

Reply:
There are two patterns AFAIK: One in [GpgVersionParser.GpgVersionConsumer](https://github.com/apache/maven-gpg-plugin/blob/master/src/main/java/org/apache/maven/plugins/gpg/GpgVersionParser.java#L113) and this one here.
This one could possibly be removed and `GpgVersionParser.GpgVersionConsumer` could pass the extracted version number to `GpgVersion`.

Reply:
It might be even better to use `gpgconf --query-swdb gnupg` instead of `gpg --version`, because gpgconf returns [machine-readable output](https://github.com/gpg/gnupg/blob/master/tools/gpgconf.c#L360).

> $ gpgconf --query-swdb gnupg
> gnupg:2.2.27:-::32849:::::::

although I don't know if that would work with all distributions.

Reply:
Works on mac, too: 
```
$ gpgconf --query-swdb gnupg
gnupg:2.2.21:-::32849:::::::
```